src/realm/query_expression.hpp

```diff
@@ -2105,13 +2111,15 @@ class Columns<Dictionary> : public ObjPropertyExpr<Mixed> {
     Columns(Columns const& other)
         : ObjPropertyExpr<Mixed>(other)
         , m_key(other.m_key)
+        , m_prop_list(other.m_prop_list)
         , m_objkey(other.m_objkey)
         , m_key_type(other.m_key_type)
     {
     }
 
 private:
     Mixed m_key;
+    std::vector<StringData> m_prop_list;
```

Comment:
can you make this a std::string so it copies in the data and then the query won't be tied to the lifetime of the search property?

Reply:
Sure.